pandasql/core.py

```diff
@@ -47,7 +50,13 @@ def __init__(self, name=None, sources=None):
     @property
     def result(self):
         if self._cached_result is None:
-            return None
+            if self._out_of_memory:
+                raise MemoryError('The result of this dataframe is too big to '
+                                  'fit in memory. Please try accessing a '
```

Comment:
Should we include a suggestion to delete stale data? Might be confusing if df.head() can't even fit in memory

Reply:
Yes good idea